### PR TITLE
chore: Bump libp2p-swarm dependents

### PR DESCRIPTION
**Pull request by mxinden**
## Description

Follow-up to https://github.com/libp2p/rust-libp2p/pull/3170#issuecomment-1344847006

## Links to any relevant issues

Should as well "fix" the cargo semver check failures for `libp2p-dcutr` and `libp2p-relay`, see https://github.com/libp2p/rust-libp2p/pull/2647#issuecomment-1335189994.

## Change checklist

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] A changelog entry has been made in the appropriate crates

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -50,7 +50,27 @@
...
+
+    - Update to [`libp2p-relay` `v0.15.0`](protocols/relay/CHANGELOG.md#0150).
+
+    - Update to [`libp2p-identify` `v0.42.0`](protocols/identify/CHANGELOG.md#0420).
+
+    - Update to [`libp2p-rendezvous` `v0.12.0`](protocols/rendezvous/CHANGELOG.md#0120).
+
+    - Update to [`libp2p-metrics` `v0.12.0`](misc/metrics/CHANGELOG.md#0120).
+
+    - Update to [`libp2p-swarm` `v0.42.0`](swarm/CHANGELOG.md#0420).
```
Comment (elenaf9) [inline CHANGELOG.md:73]:
```suggestion
    - Update to [`libp2p-swarm` `v0.42.0`](swarm/CHANGELOG.md#0420).
    
    - Update to [`libp2p-mdns` `v0.43.0`](protocols/mdns/CHANGELOG.md#0430).
    
```

**Review by elenaf9 — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -50,7 +50,30 @@
...
+    
+    - Update to [`libp2p-mdns` `v0.43.0`](protocols/mdns/CHANGELOG.md#0430).
```
Comment (mxinden) [inline CHANGELOG.md:75]:
```suggestion
    - Update to [`libp2p-mdns` `v0.43.0`](protocols/mdns/CHANGELOG.md#0430).
    
    - Update to [`libp2p-gossipsub` `v0.44.0`](protocols/gossipsub/CHANGELOG.md#0440).
```

**Review by thomaseizinger — APPROVED** [review]
I'd like us to start prioritizing, which changes are worth a breaking change. Removing deprecated APIs is IMO not worth it, the motivation should be bigger. We should obviously remove them eventually but it would be great if we can prioritize merging backwards-compatible changes and make more patch releases.

#3170 is already merged so I am happy to go ahead here but it is something to consider for the future.

**Comment by thomaseizinger** [comment]
> Should as well "fix" the cargo semver check failures for `libp2p-dcutr` and `libp2p-relay`, see [#2647 (comment)](https://github.com/libp2p/rust-libp2p/pull/2647#issuecomment-1335189994).

Only for the current release cycle though. Once it is released, it will come back unless we also merge https://github.com/libp2p/rust-libp2p/pull/3213.

**Comment by mxinden** [comment]
> I'd like us to start prioritizing, which changes are worth a breaking change. Removing deprecated APIs is IMO not worth it, the motivation should be bigger. We should obviously remove them eventually but it would be great if we can prioritize merging backwards-compatible changes and make more patch releases.

That is a good point I did not consider before.

Thus far I was operating with the mindset of always-remove-deprecation-in-next-release. Given this more thought, there is no good reason to do so, i.e. removing deprecated methods is not worth a breaking change in itself.

> #3170 is already merged so I am happy to go ahead here but it is something to consider for the future.

:+1: please continue being vocal about this. I will try to do better in the upcoming releases.

**Comment by thomaseizinger** [comment]
> > #3170 is already merged so I am happy to go ahead here but it is something to consider for the future.
> 
> :+1: please continue being vocal about this. I will try to do better in the upcoming releases.

It only came to my mind on this PR actually! To make this planning manageable, I'd suggest we create milestones. We can add open PRs to that so we know what to merge for the next release. We might also want to think about a policy for how long we keep deprecated APIs around f.e. (1, 2, ... n releases?)

**Pull request merged**